cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -196,11 +202,20 @@ public KafkaReconciler(
         this.operationTimeoutMs = config.getOperationTimeoutMs();
         this.kafkaNodePoolCrs = nodePools;
 
+        this.kafkaMetadataStateManager = new KafkaMetadataStateManager(reconciliation, kafkaCr);
+        KafkaMetadataStateManager.KafkaMetadataConfigurationState kafkaMetadataConfigState = this.kafkaMetadataStateManager.computeMetadataConfigurationState();
```

Comment:
Imagine how many characters you save in all the code if `KafkaMetadataConfigurationState` was a separate class. The code will be a lot more readable.

Reply:
You can't imagine how much I was struggling and debating with myself on this, the code looks horrible and too long to me as well. My thought was trying to have it as an inner class because strictly related to the state manager. I am going to move it out as a separate class, at least I will have approval from one of the maintainers to do it :-)

Reply:
I would probably split it. But I guess you could also consider renaming it if you don't want to split it. You for example do not need to have the `KafkaMetadata` twice there if you keep it as an inner class.

Reply:
I decided to split it. Let's see if it's better. We can reconsider later or if others will have a different comment on this.